### ISSUE TITLE
Arrow functions should not have a 'prototype' property

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#1761]
``` TypeScript
function f() {
    var a = () => 5;
    return a.prototype;
}
```

We should give an error here.

**Comment by RyanCavanaugh** [comment]
I think we have no way to do this? The apparent type of a callable type includes the members from the global type `Function` and we can't really remove `prototype` from that declaration.

**Comment by mhegazy** [comment]
The only way would be to have lambda functions have a different apparent type from regular functions.

**Comment by RyanCavanaugh** [comment]
No one else has noticed in 4 years
